src/tree_search.cpp

```diff
@@ -341,15 +341,15 @@ std::unique_ptr<Node> find_best_split(const std::vector<flat_set>& sorted_sets,
       return level_zero_learning(sorted_sets, data);
     } else {
           // "pruning", the recursive case (same action in both leaves):
-      if ((best_left_child->action_id != -1) &&
+      if ((best_left_child->left_child == nullptr && best_right_child->left_child == nullptr) &&
               (best_left_child->action_id == best_right_child->action_id)) {
         double leaf_reward = best_left_child->reward + best_right_child->reward;
         size_t leaf_action = best_left_child->action_id;
-        best_ans_as_leaf = std::unique_ptr<Node> (new Node(-1, -1, leaf_reward, leaf_action));
+        best_ans_as_leaf = std::unique_ptr<Node> (new Node(0, -1, leaf_reward, leaf_action));
```

Comment:
There are still other places to make this change: like :195

Reply:
And please add a more descriptive PR title and description

Reply:
Upcoming..